test/test_calendar.py

```diff
@@ -150,3 +156,107 @@ def test_parse_date_offset_no_offset_magnitude():
 
     expected_error = "No numerical value given for offset: YS"
     assert str(exc_info.value) == expected_error
+
+
+@pytest.mark.parametrize(
+        "start_date, end_date, caltype_int, expected",
+        [
+            (
+                datetime.datetime(year=4, month=1, day=1),
+                datetime.datetime(year=5, month=1, day=1),
+                GREGORIAN,
+                366 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=4, month=1, day=1),
+                datetime.datetime(year=5, month=1, day=1),
+                NOLEAP,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=300, month=1, day=1),
+                datetime.datetime(year=301, month=1, day=1),
+                GREGORIAN,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=400, month=1, day=1),
+                datetime.datetime(year=401, month=12, day=31),
+                GREGORIAN,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=12, month=7, day=22),
+                datetime.datetime(year=23, month=3, day=15),
+                GREGORIAN,
+                (10 * 365 + 238) * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=1, month=1, day=1),
+                datetime.datetime(year=1001, month=1, day=1),
+                GREGORIAN,
+                (1000 * 365 + 242) * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=1, month=1, day=1),
+                datetime.datetime(year=9999, month=1, day=1),
+                GREGORIAN,
+                (9998 * 365 + 2424) * SEC_PER_DAY
+            )
+        ]
+)
+def test_seconds_between_dates(start_date, end_date, caltype_int, expected):
+    assert seconds_between_dates(start_date, end_date, caltype_int) == expected
+
+
+@pytest.mark.parametrize(
+        "date_int, expected",
+        [
+            (10101, datetime.date(1, 1, 1)),
+            (100321, datetime.date(10, 3, 21)),
+            (38211130, datetime.date(3821, 11, 30))
+        ]
+)
+def test_int_to_date(date_int, expected):
+    """
+    Check that integers typically read in from namelists
+    are correctly converted to datetime.date objects.
+    """
+    converted_date = int_to_date(date_int)
+    assert converted_date == expected
+
```

Comment:
For completeness, this should also test the negative cases:

e.g. what if the year is 0 or year 10000, or the date requested doesn't exist (i.e. 999999)

Reply:
Good question, I've added a separate test to check that the function fails on invalid dates:

https://github.com/ACCESS-NRI/payu/blob/c19133c83109e768236486b0879d09e6f05c3be2/test/test_calendar.py#L229-L239

Reply:
Nice